### PR TITLE
Feature add signup

### PR DESCRIPTION
**Pull request by ArsalanSavand**

### CODE REVIEW

**Comment by amirm3hdi** [comment]
@ArsalanSavand You must follow the template rules to not to change layout style

**Comment by Alir3z4** [comment]
@ArsalanSavand Is this ready to be merged?
If so I can merge it right away.

**Comment by amirm3hdi** [comment]
@Alir3z4 No, I'm going to make changes.
Also the registration doesn't submit.

File: app/signup/signup_controller.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    // Signup form
+    $scope.form = {};
+
+    /**
+     * constructor
+     *
+     * @method constructor
+     * @desc Init function for controller
+     *
+     * @memberOf RegisterController
```
Comment (amirm3hdi) [inline app/signup/signup_controller.js]: Documentation is wrong.

File: app/signin/signin_view.html [hunk 2]
```diff
@@ -15,9 +15,11 @@ <h3 class="o-fade-soft">Login</h3>
                     <div class="form-group alert alert-danger" ng-show="form.errors.non_field_errors">
                         {{ form.errors.non_field_errors[0] }}
                     </div>
-                    <button class="btn btn-primary pull-right btn-lg" ng-disabled="form.loading"><i class="fa fa-sign-in fa-fw"></i></button>
```
Comment (amirm3hdi) [inline app/signin/signin_view.html]: No centering and changing the layout.

File: app/signup/signup_controller.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    };
+
+    /**
+     * signup
+     *
+     * @method signup
+     * @desc Submit signup form to authenticate
+     *
+     */
+     $scope.signup = function register(form) {
```
Comment (amirm3hdi) [inline app/signup/signup_controller.js]: This is wrong completely.

**Comment by Alir3z4** [comment]
@amirm3hdi @ArsalanSavand I'm not sure why the tests failed.
You can run the tests and see what's going on:
```
grunt test
```

Tests are not important right now at this moment, but the structure and basics of it should kept going just fine so later we can write some tests for the app without having to install everything

File: app/signup/signup_view.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<div class="container container-light pad-top">
+    <div class="col-md-4 col-md-offset-4">
+        <div class="panel panel-default">
+            <div class="panel-body s-pad s-box">
+                <form ng-submit="signup(form)" name="form">
+                    <h3 class="o-fade-soft">Register</h3>
+                    <hr>
+                    <div class="form-group" ng-class="{'has-error': form.errors.email}">
+                        <input id="email" class="form-control input-lg" type=email ng-model="form.email" placeholder="Email" required/>
```
Comment (amirm3hdi) [inline app/signup/signup_view.html]: No quotations around the type attr.

File: app/signup/signup_controller.js (hunk 1 above)
Comment (Alir3z4) [inline app/signup/signup_controller.js]: @amirm3hdi You need to also let @ArsalanSavand know how to fix it as well, suggest him how to fix it please.

**Comment by Alir3z4** [comment]
@ArsalanSavand You can find the ngdocs standard here: https://github.com/angular/angular.js/wiki/Writing-AngularJS-Documentation

**Pull request merged**